.github/workflows/code-quality.yml

```
@@ -0,0 +1,84 @@
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'npm'
+
+      - name: Install WebUI dependencies
+        run: cd src/app/webui && npm ci
+
+      - name: TypeCheck Frontend (via build)
+        run: cd src/app/webui && npm run build
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)